arviz/plots/traceplot.py

```diff
@@ -97,11 +100,13 @@ def plot_trace(
     """
     if divergences:
         try:
-            divergence_data = convert_to_dataset(data, group="sample_stats").diverging
+            divergence_data = get_coords(
+                convert_to_dataset(data, group="sample_stats"), coords
+            ).diverging
```

Comment:
This is not good and there are many possible solutions, however, I feel like the cleanest one would be:

```python
""" #docstring end
data = convert_to_inference_data(data)
data = data.sel(inplace=False, **coords)
```
So that afterwards, the posterior and the divergence data are already retrieved with the correct coords.

Reply:
But it requires too much memory though, as the whole inference data object is copied. Open to suggestions.